crypto/fipsmodule/rand/asm/rndr-armv8.pl

```diff
@@ -0,0 +1,75 @@
+#! /usr/bin/env perl
+
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0 OR ISC
+
+# RNDR from ARMv8.5-A.
+# System register encoding: s3_3_c2_c4_0
+# see https://developer.arm.com/documentation/ddi0601/2024-09/AArch64-Registers/RNDR--Random-Number
+
+# The first two arguments should always be the flavour and output file path.
+if ($#ARGV < 1) { die "Not enough arguments provided.
+  Two arguments are necessary: the flavour and the output file path."; }
+
+my $flavour = shift;
+my $output  = shift;
+
+$0 =~ m/(.*[\/\\])[^\/\\]+$/; $dir=$1;
+( $xlate="${dir}arm-xlate.pl" and -f $xlate ) or
+( $xlate="${dir}../../../perlasm/arm-xlate.pl" and -f $xlate) or
+die "can't locate arm-xlate.pl";
+
+open OUT,"| \"$^X\" $xlate $flavour $output";
+*STDOUT=*OUT;
+
+$code.=<<___;
+#include <openssl/arm_arch.h>
+
+.arch armv8-a
+.text
+
+# int CRYPTO_rndr(uint8_t *out, const size_t len)
+.globl CRYPTO_rndr
+.type CRYPTO_rndr,%function
+.align 4
+CRYPTO_rndr:
+  cbz x1, .Lrndr_error        // len = 0 is not supported
+  mov x4, x1                  // len: requested number of bytes
+  mov x2, #0                  // Counts number of bytes generated
+
+.Lrndr_loop:
+  mrs x3, s3_3_c2_c4_0        // rndr instruction
+  cbz x3, .Lrndr_error        // Check if RNDR failed
+
+  cmp x1, #8                  // Sets N if strictly less than 8 bytes left
```

Comment:
what's `N`?

Reply:
Condition flags. Input are to be interpreted as unsigned though, so should be checking for `C` being set and use `blo`. The case of `C` not being set is the case where `x1 < 8` and we need to handle bit-by-bit.
I changed that.